### DISCUSSION TITLE
Replacing a button with custom attributes

### DISCUSSION THREAD

**Discussion by helloingob** [remarkablemark/html-react-parser#267] [General]
Hey,

I got a html string from an API and I want to replace some attributes. The attribute "**method**" and "**param**" should be **removed** and used for "**onClick**" and the **rest** of the attributes should be **kept**.

**Example input string**
```html
<button method="SetColor" param="#FFFFFF" class="abc" id="btn1">ABC</button>
```
**React**
```javascript
{Parser(props.content,  {
  replace: domNode => {
    if (domNode.name === 'button' && domNode.attribs && domNode.attribs.method !== '' && domNode.attribs.param !== '')  {
        return <button onClick={() => sendCommand(domNode.attribs.method, domNode.attribs.param)}>XYZ</button>;
    }
  }
})}
```

This code is working **great**, but how do I keep **other** attributes like "**class**" or "**id**"? And how do I access the button's name "**ABC**" and print it instead of "**YXZ**". I know "[_innerText_](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText)", but it's not valid.

How can I solve this?
Any help greatly appreciated :)
Thanks

**Comment by remarkablemark** [comment]
Can you use destructuring assignment? For example:

```jsx
{Parser(props.content,  {
  replace: domNode => {
    if (domNode.name === 'button' && domNode.attribs && domNode.attribs.method !== '' && domNode.attribs.param !== '')  {
        // see object destructuring below
        const { method, param, ...restProps } = domNode.attribs;
        return <button {...restProps} onClick={() => sendCommand(method, param)}>XYZ</button>;
    }
  }
})}
```
